### DISCUSSION TITLE
Calculate number of hours/days/months between two dates

### DISCUSSION THREAD

**Discussion by hhamzic1** [OData/AspNetCoreOData#856] [Q&A]
Greetings!

I've stumbled on one issue that I couldn't solve when I tried to calculate basically _**datediff**_ between _**current time**_ (now()) and some **_DateTime_** object.

What I want to do is to calculate _**number of seconds**_ that passed between `now()` and some `DateTime` object. I've struggled with this one and the 'best' result that I was able to achieve was with this query:

`?$compute=cast(now() sub CreatedAt, Edm.String) as DateDiff&$select=DateDiff`

and got these results:

```
"value": [
        {
            "DateDiff": "2 days 19:31:48.177806"
        },
        {
            "DateDiff": "2 days 18:40:32.254084"
        },
        {
            "DateDiff": "16:33:13.852226"
        },
        {
            "DateDiff": "16:31:26.543277"
        }]
```
        
the problem with this is that I need a **number** as a result that represents e.g. number of seconds/hours/days between the dates so that I could use it to divide some other number with it. 

Is there a solution for this?

**Comment by habbes** [comment]
There's a `totalseconds` built-in query function defined in the spec: https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_BuiltinQueryFunctions

The input of the function is a duration. Could you try `$compute=totalseconds(now() sub CreatedAt)` and see if that works?

**Reply by hhamzic1** [reply]
@habbes Unfortunately it says this:

`"message": "The query specified in the URI is not valid. Unknown function 'totalseconds'.",`

**Comment by ivanovvitaly** [comment]
Hi, when `totalseconds` and `totaloffsetminutes` will be implemented? I'm using `Microsoft.AspNetCore.OData 9.2.1` and it still lacking those functions.
https://github.com/OData/AspNetCoreOData/blob/699f0533eff09396eb30503c56d95fdb3a172412/src/Microsoft.AspNetCore.OData/Query/ClrCanonicalFunctions.cs

Is there a way to implement these functions on my own? Give me a hint how to extend the framework
